enum_methods.rb

```diff
@@ -0,0 +1,146 @@
+module Enumerable
+  def my_each
+    if block_given?
+      for i in self
+        yield(i)
+      end
+    else
+      puts 'No Block Given'
+    end
+  end
+
+  def my_each_with_index
+    count = 0
+    if block_given?
+      for i in self
+        yield(i, count)
+        count += 1
+      end
+    else
+      puts 'No Block Given'
+    end
+  end
+
+  def my_select
+    if instance_of?(Hash)
+      result = {}
+      my_each { |key, value| result[key] = value if yield(key, value) == true }
+    else
+      result = []
+      my_each { |value| result.push(value) if yield(value) == true }
+    end
+    result
+  end
+
+  def my_all?
+    count = 0
+    my_each { |value| count += 1 if yield(value) == true }
+    count == length
+  end
+
+  def my_any?
+    count = 0
+    my_each { |value| break count += 1 if yield(value) == true }
+    count.positive?
+  end
+
+  def my_none?
+    count = 0
+    my_each { |value| break count += 1 if yield(value) == true }
+    !count.positive?
+  end
+
+  def my_count
+    count = 0
+    if block_given?
+      my_each do |value|
+        count += 1 if yield(value) == true
+      end
+      count
+    else
+      puts 'No Block Given'
+    end
+  end
+
+  def my_map(proc = nil)
+    result = []
+    if proc.nil?
+      if block_given?
+        my_each { |key, value| result.push(yield(key, value)) }
+        result
+      else
+        self
+      end
+    else
+      my_each { |key, value| result.push(proc.call(key, value)) }
+      result
+    end
+  end
+
+  def my_inject(arg = self[0])
+    count = 1
+    if arg == self[0]
+      my_each do |_value|
+        if count < length
+          arg = yield(arg, self[count])
+          count += 1
+        end
+      end
+    else
+      my_each { |value| arg = yield(arg, value) }
+    end
+    arg
+  end
+end
```

Comment:
- [ ] my_inject when a block is given without an initail value combines all elements of enum by applying a binary operation, specified by a block::range

- [ ] my_inject when a block is given with an initail value combines all elements of enum by applying a binary operation, specified by a block

- [ ] my_inject when a symbol is specified without an initail value combines each element of the collection by applying the symbol as a named method

- [ ] my_inject when a symbol is specified without an initail value combines each element of the collection by applying the symbol as a named method:range

- [ ] my_inject when a symbol is specified with an initail value combines each element of the collection by applying the symbol as a named method

- [ ] my_inject when a symbol is specified with an initail value combines each element of the collection by applying the symbol as a named method::range